app/models/docker_builder_service.rb

```diff
@@ -11,6 +11,8 @@ def initialize(build)
   end
 
   def run!(image_name: nil, push: false)
+    build.docker_build_job.try(:destroy) # if there's an old build job, delete it
```

Comment:
`DockerBuilderService` needs more tests in general, since the `run!` method currently doesn't have any tests.  I'll circle back on this next week